app/helpers/ExerciseConfig/EntityMetadata/HwGroupMeta.php

```diff
@@ -0,0 +1,135 @@
+  private static $WALL_TIME_PER_EXERCISE_KEY = "wallTimePerExerciseKey";
+
+
+  /**
+   * Maximal memory limit for hwgroup in kilobytes.
+   * @var int
+   */
+  private $memory;
+
+  /**
+   * Maximal cpu-time for hwgroup per test in seconds.
+   * @var float
+   */
+  private $cpuTimePerTest;
+
+  /**
+   * Maximal cpu-time for hwgroup per exercise in seconds.
+   * @var float
+   */
+  private $cpuTimePerExercise;
+
+  /**
+   * Maximal wall-time for hwgroup per test in seconds.
+   * @var float
+   */
+  private $wallTimePerTest;
+
+  /**
+   * Maximal wall-time for hwgroup per exercise in seconds.
+   * @var float
+   */
+  private $wallTimePerExercise;
+
+
+  /**
+   * HwGroupMeta constructor.
+   * @param string $data
+   */
+  public function __construct(string $data) {
+    $parsed = Yaml::parse($data);
+
+    if (array_key_exists(self::$MEMORY_KEY, $parsed)) {
+      $this->memory = $parsed[self::$MEMORY_KEY];
+    }
+
+    if (array_key_exists(self::$CPU_TIME_PER_TEST_KEY, $parsed)) {
+      $this->cpuTimePerTest = $parsed[self::$CPU_TIME_PER_TEST_KEY];
+    }
+
+    if (array_key_exists(self::$CPU_TIME_PER_EXERCISE_KEY, $parsed)) {
+      $this->cpuTimePerExercise = $parsed[self::$CPU_TIME_PER_EXERCISE_KEY];
+    }
+
+    if (array_key_exists(self::$WALL_TIME_PER_TEST_KEY, $parsed)) {
+      $this->wallTimePerTest = $parsed[self::$WALL_TIME_PER_TEST_KEY];
+    }
+
+    if (array_key_exists(self::$WALL_TIME_PER_EXERCISE_KEY, $parsed)) {
+      $this->wallTimePerExercise = $parsed[self::$WALL_TIME_PER_EXERCISE_KEY];
+    }
+  }
+
+
+  /**
+   * Maximal memory limit for hwgroup in kilobytes.
+   * @return int
+   */
+  public function getMemory(): int {
+    return $this->memory;
+  }
+
+  /**
+   * Maximal cpu-time for hwgroup per test in seconds.
+   * @return float
+   */
+  public function getCpuTimePerTest(): float {
+    return $this->cpuTimePerTest;
+  }
+
+  /**
+   * Maximal cpu-time for hwgroup per exercise in seconds.
+   * @return float
+   */
+  public function getCpuTimePerExercise(): float {
+    return $this->cpuTimePerExercise;
+  }
+
+  /**
+   * Maximal wall-time for hwgroup per test in seconds.
+   * @return float
+   */
+  public function getWallTimePerTest(): float {
+    return $this->wallTimePerTest;
+  }
+
+  /**
+   * Maximal wall-time for hwgroup per exercise in seconds.
+   * @return float
+   */
+  public function getWallTimePerExercise(): float {
+    return $this->wallTimePerExercise;
+  }
+
+
+  /**
+   * Serialize hardware group meta information into array.
+   * @return array
+   */
+  public function toArray(): array {
+    return [
+      self::$MEMORY_KEY => $this->memory,
+      self::$CPU_TIME_PER_TEST_KEY => $this->cpuTimePerTest,
+      self::$CPU_TIME_PER_EXERCISE_KEY => $this->cpuTimePerExercise,
+      self::$WALL_TIME_PER_TEST_KEY => $this->wallTimePerTest,
+      self::$WALL_TIME_PER_EXERCISE_KEY => $this->wallTimePerExercise
+    ];
```

Comment:
Similarly to my previous remark, this could have been done by foreach/array_map.